memory/table.go

```diff
@@ -1002,6 +1120,29 @@ func (t *IndexedTable) LookupPartitions(ctx *sql.Context, lookup sql.IndexLookup
 		return nil, err
 	}
 
+	if lookup.Index.IsSpatial() {
+		lower := sql.GetRangeCutKey(lookup.Ranges[0][0].LowerBound)
+		upper := sql.GetRangeCutKey(lookup.Ranges[0][0].UpperBound)
+		minPoint, ok := lower.(types.Point)
+		if !ok {
+			return nil, sql.ErrInvalidGISData.New()
+		}
+		maxPoint, ok := upper.(types.Point)
+		if !ok {
+			return nil, sql.ErrInvalidGISData.New()
+		}
+
+		ord := lookup.Index.(*Index).Exprs[0].(*expression.GetField).Index()
```

Comment:
do spatial indexes support nary columns?

Reply:
Spatial indexes can only be over exactly one geometry column
[MySQL docs](https://dev.mysql.com/doc/refman/8.0/en/create-index.html#create-index-spatial:~:text=Available%20for%20single%20spatial%20columns%20only.%20A%20spatial%20index%20cannot%20be%20created%20over%20multiple%20spatial%20columns.)

Reply:
in that case, would it make more sense to have a separate SpatiaIndexBuilder?